esmf_regrid/esmf_regridder.py

```diff
@@ -176,12 +176,17 @@ def regrid(self, src_array, norm_type="fracarea", mdtol=1):
         extra_size = max(1, np.prod(extra_shape))
         src_inverted_mask = self.src._array_to_matrix(~ma.getmaskarray(src_array))
         weight_sums = self.weight_matrix @ src_inverted_mask
+        src_dtype = src_array.dtype
+        if np.issubdtype(src_dtype, np.floating):
+            out_dtype = src_dtype
+        else:
+            out_dtype = np.float64
```

Comment:
Would it be safer here to use the weights.dtype, if that is float, and only if _that_ is integral default to a fixed floating precision ?
Otherwise, **_any_** integer data will be done in float64, when the weights array might indicate that a lower precision was "good enough" (and presumably cheaper).

Reply:
I believe the weights tend to be float64 by default, but I suppose this could change if you're using user supplied weights so this is probably worth implementing.

Reply:
Ok, done!